balloc/balloc.c

```diff
@@ -706,13 +717,18 @@ static void balloc_group_write_do(struct m0_be_tx_bulk *tb,
 	spare_size = m0_stob_ad_spares_calc(sb->bsb_groupsize);
 	/* Insert non-spare extents. */
 	ext.e_start = i << sb->bsb_gsbits;
-	ext.e_end = ext.e_start + sb->bsb_groupsize - spare_size;
+	ext.e_end = ext.e_start + sb->bsb_groupsize - spare_size - 1;
```

Comment:
This was a bug in the original code?

Reply:
Looks like. this has been observe when i tried to split device slices based on no of group

Reply:
The ext is [ext.e_start, ext.e_end)

The end is not included in the extent. 
Please confirm.

Reply:
The end is the start addr. of the next extent but also happens to be the key of the current extent.. Since ext.e_end is used as the key a few lines below the key (e_end) should be the start of the next extents start addr hence we do not need to have '-1' here.

Reply:
I will take care this in next push.